### DISCUSSION TITLE
Using Textual on Linux console

### DISCUSSION THREAD

**Discussion by twdav** [Textualize/textual#2902] [Q&A] [answered]
Hi all - I think I know the answer to this, but we would like to use the fantastic Textual tool on a network appliance running RHEL9 that we're building and have questions.

We've run into degradation in appearance on a linux console vs. in a terminal application on a Gnome desktop. Also, things don't render well from a Windows machine in Putty either, but that's not really an issue as Windows terminal could be used and works perfectly.

I'm assuming this is b/c the console simply does not have the capabilities of the terminal application, and this is how it will be. But, perhaps there's something I'm missing or some alternate paths we could take to improve the appearance on the console?

Gnome

![PXL_20230707_180956094](https://github.com/Textualize/textual/assets/138918981/d135f1df-8165-4de5-bfbc-506c4c486e84)

Console

![PXL_20230707_181011303](https://github.com/Textualize/textual/assets/138918981/2d1ecaff-fadd-4462-a826-7dcd60f4cde9)

**Comment by willmcgugan** [comment, marked as answer]
We would like to help you with this use-case. I suspect it could be improved with some tweaking. Jump on the Discord server if you would like to discuss it!

**Reply by twdav** [reply]
Thanks @willmcgugan I will get in touch there, thanks.

**Comment by nicko27** [comment]
Hi, I'm interested by the answer as I am in the same situation. Please

**Comment by wilrodriguez** [comment]
I also have this use-case. Our application is a system configuration wizard that runs on first boot for our platform automation and... well... we don't really want to run in graphical mode if we can help it, since our app runs in k8s, we try and keep the underlying OS minimal for security purposes.

**Reply by willmcgugan** [reply]
If its a commercial use case, I can help in return for a sponsorship. https://github.com/sponsors/willmcgugan/

**Reply by wilrodriguez** [reply]
I can't promise I can get the company to sponsor, but I'll gladly throw in my own cash. Textual is such an awesome framework and I really want to see what the future holds for it... even if I have to put my own project on ice.

**Comment by Krovavii** [comment]
Yes, you can. It's working fine if you change CSS of all widgets. You just need change any borders style to 'solid' or 'none'. Also you can use only 8 basic colors (not sure, but: white, black, grey, green, yellow, blue, orange and purple). Unfortunately you can't avoid these restiction in TTY.

**Comment by twdav** [comment]
If it helps, we ended up accommodating this limitation by adding a check for the available color space:

```
def is_rich_terminal():
    """
    Determine if the environment supports a 256-color terminal.
    """
    term = os.getenv("TERM", "")
    return "256" in term
```

and then loading different stylesheets depending on the result of that check. The 16-color terminal stylesheet does largely what @Krovavii suggests, simplifying borders etc. to use characters available in the screen font and limiting color use. It works well. 

Things still retain full functionality when running in an environment which permits it by using the 'full' stylesheet.

**Reply by edward-jazzhands** [reply]
@twdav The idea of having a mode for doing this built directly into Textual has actually come up in the Discord server before. I could imagine this being a future feature if someone was to take the time to make it

**Reply by twdav** [reply]
I think that would make sense. As it turns out, there are quite a few things to take into account. 

Since we know which distro (including screen font) our app will run on, a simple check like the above works. But to make it work across multiple distros and taking into account OS installation quirks would be a bigger issue. Different screen fonts have different glyphs, for example, and might be able to support a greater portion of the capability of a full terminal font (e.g., TrueType) as far as rendering the Textual interface components.

**Reply by edward-jazzhands** [reply]
@twdav A simple solution to that might be to make the mode itself be something built into Textual with a simple toggle (reactive variable on App class) and yet have the logic to activate it still need to be implemented by the programmer, so Textual makes absolutely no attempt to guess that for you.
